python/src/aac/lang/definitions/extensions.py

```diff
@@ -0,0 +1,83 @@
+"""Module for handling Definitions and Extensions."""
+
+import logging
+
+from aac.lang.definitions.definition import Definition
+
+
+def apply_extension_to_definition(extension_definition: Definition, target_definition: Definition) -> None:
+    """
+    Apply the extension to the definition it it modifies.
+
+    Args:
+        extension_definition (Definition): The extension definition to apply to the context.
+        target_definition (Definition): The extension definition to apply to the context.
+    """
+    target_definition_extension_sub_dict = target_definition.get_fields()
+    extension_subtype_sub_dict = _get_extension_subtype_dict(extension_definition)
```

Comment:
I've made the method name a little clearer in the data -> schema PR: https://github.com/jondavid-black/AaC/blob/a0d2654d1cf6b7e5c46bea54838c227570e3becb/python/src/aac/lang/definitions/definition.py#L30-L38 

The confusion is the schema/data has a 'fields' field that defines everyone's list of top-level fields. I though that I had expanded on the function description in that PR to clarify that issue, but it may not have been saved.